rspec-core/lib/rspec/core/configuration.rb

```diff
@@ -1765,14 +1766,57 @@ def self.delegate_to_ordering_manager(*methods)
 
       # Set Ruby warnings on or off.
       def warnings=(value)
-        $VERBOSE = !!value
+        @warnings_set = true
+
+        case value
+        when true, false
+          RSpec.deprecate(
+            "RSpec::Core::Configuration#warnings=",
+            :message =>  "config.warnings = #{value} is deprecated, please set warnings to one of: " \
+                         "`:none`, `:all` or `:deprecations_only.`"
+          )
+          value = value ? :all : :none
+        end
+
+        case value
+        when :none
+          $VERBOSE = false
+          self.deprecation_warnings = false
+        when :all
+          $VERBOSE = true
+          self.deprecation_warnings = true
+        when :deprecations_only
+          $VERBOSE = false
+          self.deprecation_warnings = true
+        else
+          raise "Unsupported value for `warnings` (#{value.inspect}). " \
```

Comment:
It's this line that decreases coverage (from an unreal 100%)

<img width="753" alt="image" src="https://github.com/user-attachments/assets/bc0c8acb-eac0-45c5-a473-2cc25d79bd62" />

Reply:
Good catch, I added a test for it.